[chore] processor/deltatocumulative: data.Adder

tombrk opened the pull request:
<!--Ex. Fixing a bug - Describe the bug and how this fixes the issue.
Ex. Adding a feature - Explain what this achieves.-->
#### Description

Introduces data.Adder, a struct that performs adding of pmetric.*DataPoint.

This opens up future features that require configuration of the adding operation, such as out of order support.

<!--Describe what testing was performed and which tests were added.-->
#### Testing
Existing tests pass unaltered

tombrk committed:
processor/deltatocumulative: data.Adder

Introduces data.Adder, a struct that performs adding of
pmetric.*DataPoint.

This opens up future features that require configuration of the adding
operation, such as out of order support.

dehaansa commented on processor/deltatocumulativeprocessor/internal/data/add.go:
```diff
@@ -12,120 +12,133 @@ import (
 	}
-	dp.SetTimestamp(in.Timestamp())
-	return dp
+	state.SetTimestamp(dp.Timestamp())
```
What would define that "adding" two data points sets the timestamp to the newer one? I think the abstraction fails here a little bit, but I don't have a clear solution.

sh0rez replied:
oh good catch .. we could move that to delta.Aggregator which arguably fits this a lot better

tombrk committed:
internal/delta: advance timestamps

moves timestamp handling from internal/data.Adder to
internal/delta.Aggregate, as it arguably belongs there.
